src/probnum/diffeq/_perturbsolve_ivp.py

```diff
@@ -0,0 +1,191 @@
+    noise_scale=10.0,
+    adaptive=True,
+    atol=1e-2,
+    rtol=1e-2,
+    step=None,
+):
+    r"""Solve an initial value problem with a perturbation-based probabilistic ODE solver.
+
+    Parameters
+    ----------
+    f :
+        ODE vector field.
+    t0 :
+        Initial time point.
+    tmax :
+        Final time point.
+    y0 :
+        Initial value.
+    rng :
+        Random number generator.
+    method :
+        Integration method to use.
+        The following are available (docs adapted from `SciPy <https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html>`_):
+
+            * `RK45` (default): Explicit Runge-Kutta method of order 5(4) [2]_.
+              The error is controlled assuming accuracy of the fourth-order
+              method, but steps are taken using the fifth-order accurate
+              formula (local extrapolation is done). A quartic interpolation
+              polynomial is used for the dense output [3]_. Can be applied in
+              the complex domain.
+            * `RK23`: Explicit Runge-Kutta method of order 3(2) [4]_. The error
+              is controlled assuming accuracy of the second-order method, but
+              steps are taken using the third-order accurate formula (local
+              extrapolation is done). A cubic Hermite polynomial is used for the
+              dense output. Can be applied in the complex domain.
+
+        Other integrators are not supported currently.
+    perturb
+        Which perturbation style to use.
+        Currently, the following methods are supported:
+
+            * `step-lognormal`: Perturbed-step (aka random time-step numerical integration) method
+              with lognormally distributed perturbation of the step-size [1]_.
+            * `step-uniform`: Perturbed-step (aka random time-step numerical integration) method
+              with lognormally distributed perturbation of the step-size [1]_.
+
+        Other integrators are not supported currently.
+    noise_scale
+        Scale of the perturbation. Optional. Default is 10.0. The magnitude of this parameter
+        significantly impacts the width of the posterior.
+    adaptive :
+        Whether to use adaptive steps or not. Default is `True`.
+    atol :
+        Absolute tolerance  of the adaptive step-size selection scheme.
+        Optional. Default is ``1e-4``.
+    rtol :
+        Relative tolerance   of the adaptive step-size selection scheme.
+        Optional. Default is ``1e-4``.
+    step :
+        Step size. If atol and rtol are not specified, this step-size is used for a fixed-step ODE solver.
+        If they are specified, this only affects the first step. Optional.
+        Default is None, in which case the first step is chosen as prescribed by :meth:`propose_firststep`.
+
+    Examples
+    --------
+    >>> from probnum.diffeq import perturbsolve_ivp
+    >>> import numpy as np
+
+    Solve a simple logistic ODE with fixed steps.
```

Comment:
Maybe you can clarify once again that a perturbed solver with uniformly perturbed steps is used. The default can't be seen in the code and I like the idea of providing a default perturbation function, but I think this should be made clear in this example.